### PR TITLE
[SLM] Support `q0f16` and `q0f32`

### PR DESCRIPTION
**Pull request by cyx-6**
This PR adds the support of `q0f16` and `q0f32`, and change `RMSNorm` to `nn.RMSNorm`.

### CODE REVIEW

File: python/mlc_chat/compiler/quantization/no_quantization.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                    The new node to replace current node.
+                """
+                if isinstance(node, (nn.Linear, nn.MultiLinear, nn.Embedding)):
+                    weight_name = f"{name}.weight"
+                    self.quant_map.param_map[weight_name] = [weight_name]
+                    self.quant_map.map_func[weight_name] = self.config.quantize_weight
+                return self.visit(name, node)
+
+        model.to(dtype=self.model_dtype)
+        mutator = _Mutator(self, quant_map)
```
Comment (junrushao) [inline python/mlc_chat/compiler/quantization/no_quantization.py:81]: Do we even need a mutator in `NoQuantize`? I suppose `model.to` has already converted the model to `model_dtype`

File: python/mlc_chat/compiler/model/llama_quantization.py [hunk 2]
```diff
@@ -39,3 +39,19 @@ def awq_quant(
...
+) -> Tuple[nn.Module, QuantizeMapping]:
+    """Quantize a Llama2 model without quantization."""
+    model: nn.Module = LlamaForCasualLM(model_config)
+    model.to(quantization.model_dtype)
+    quant_map = QuantizeMapping({}, {})
+    model = quantization.quantize_model(
+        model,
+        quant_map,
+        "",
+    )
```
Comment (junrushao) [inline python/mlc_chat/compiler/model/llama_quantization.py:56]: Is it possible to simply skip `quantization.quantize_model` and use an empty `QuantizeMapping`?

File: python/mlc_chat/compiler/quantization/no_quantization.py (hunk 1 above)
Comment (cyx-6) [reply python/mlc_chat/compiler/quantization/no_quantization.py]: yes, removed.

File: python/mlc_chat/compiler/model/llama_quantization.py (hunk 2 above)
Comment (cyx-6) [reply python/mlc_chat/compiler/model/llama_quantization.py]: yes, removed.

**Review by junrushao — APPROVED** [review]

**Pull request merged**